kube-client/src/client/auth/mod.rs

```diff
@@ -91,6 +93,12 @@ pub enum Error {
     #[cfg_attr(docsrs, doc(cfg(feature = "oauth")))]
     #[error("failed OAuth: {0}")]
     OAuth(#[source] OAuthError),
+
+    /// OIDC error
+    #[cfg(feature = "oidc")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "oidc")))]
+    #[error("failed OIDC: {0}")]
+    Oidc(#[source] OidcError),
```

Comment:
one minor point here, this `OidcError` is not actually exposed in the docs (can be verified by running `just doc` and browsing). i tried adding some stuff to ensure the `oidc` features got filled in everywhere in the docs with:

```diff
diff --git a/justfile b/justfile
index 3bc6c6d..9a8d6f5 100644
--- a/justfile
+++ b/justfile
@@ -15,7 +15,7 @@ fmt:
   rustfmt +nightly --edition 2021 $(find . -type f -iname *.rs)
 
 doc:
-  RUSTDOCFLAGS="--cfg docsrs" cargo +nightly doc --lib --workspace --features=derive,ws,oauth,jsonpatch,client,derive,runtime,admission,k8s-openapi/v1_26,unstable-runtime --open
+  RUSTDOCFLAGS="--cfg docsrs" cargo +nightly doc --lib --workspace --features=derive,ws,oauth,oidc,jsonpatch,client,derive,runtime,admission,k8s-openapi/v1_26,unstable-runtime --open
 
 deny:
   # might require rm Cargo.lock first to match CI
diff --git a/kube-client/Cargo.toml b/kube-client/Cargo.toml
index 035cb7c..e391dad 100644
--- a/kube-client/Cargo.toml
+++ b/kube-client/Cargo.toml
@@ -32,7 +32,7 @@ config = ["__non_core", "pem", "dirs"]
 __non_core = ["tracing", "serde_yaml", "base64"]
 
 [package.metadata.docs.rs]
-features = ["client", "rustls-tls", "openssl-tls", "ws", "oauth", "jsonpatch", "admission", "k8s-openapi/v1_26"]
+features = ["client", "rustls-tls", "openssl-tls", "ws", "oauth", "oidc", "jsonpatch", "admission", "k8s-openapi/v1_26"]
 # Define the configuration attribute `docsrs`. Used to enable `doc_cfg` feature.
 rustdoc-args = ["--cfg", "docsrs"]
 
diff --git a/kube-client/src/client/auth/mod.rs b/kube-client/src/client/auth/mod.rs
index afed09e..a218093 100644
--- a/kube-client/src/client/auth/mod.rs
+++ b/kube-client/src/client/auth/mod.rs
@@ -4,7 +4,6 @@ use std::{
     sync::Arc,
 };
 
-
 use chrono::{DateTime, Duration, Utc};
 use futures::future::BoxFuture;
 use http::{
diff --git a/kube-client/src/client/mod.rs b/kube-client/src/client/mod.rs
index c9e3cec..5b45281 100644
--- a/kube-client/src/client/mod.rs
+++ b/kube-client/src/client/mod.rs
@@ -46,6 +46,9 @@ pub use tls::openssl_tls::Error as OpensslTlsError;
 #[cfg(feature = "oauth")]
 #[cfg_attr(docsrs, doc(cfg(feature = "oauth")))]
 pub use auth::OAuthError;
+#[cfg(feature = "oidc")]
+#[cfg_attr(docsrs, doc(cfg(feature = "oidc")))]
+pub use auth::OidcError;
 
 #[cfg(feature = "ws")] pub use upgrade::UpgradeConnectionError;
 
diff --git a/kube/Cargo.toml b/kube/Cargo.toml
index 485a32f..7aea406 100644
--- a/kube/Cargo.toml
+++ b/kube/Cargo.toml
@@ -32,7 +32,7 @@ runtime = ["kube-runtime"]
 unstable-runtime = ["kube-runtime/unstable-runtime"]
 
 [package.metadata.docs.rs]
-features = ["client", "rustls-tls", "openssl-tls", "derive", "ws", "oauth", "jsonpatch", "admission", "runtime", "k8s-openapi/v1_26", "unstable-runtime"]
+features = ["client", "rustls-tls", "openssl-tls", "derive", "ws", "oauth", "oidc", "jsonpatch", "admission", "runtime", "k8s-openapi/v1_26", "unstable-runtime"]
 # Define the configuration attribute `docsrs`. Used to enable `doc_cfg` feature.
 rustdoc-args = ["--cfg", "docsrs"]
 
```

but that exposes some missing_docs via the `OidcError` enums that get exposed as a result of these changes.

I suspect that is a minor thing, but it would be good to expose this as it would be inline with the rest and it looks like an accident.

Reply:
Should be fixed now. When browsing the docs with exposed `OidcError`, I also noticed that its variants contained private types. I moved all the errors to an `errors` submodule and exposed it in `kube-client` to be concise.

Reply:
That looks sensible now. Everything about oidc implementation is private (except the errors, which kind of have to be public). Thank you!

Reply:
Although this does mean we have a different way of doing it in `oidc` compared to oauth;

![Uploading Screenshot 2023-06-30 at 11.24.00.png…]()

Reply:
I guess the alternative would be to keep the `oidc_error` module private, but then we would have to make all the error types in oidc errors re-exposed from client/mod 🤔

Reply:
I think this is probably the best we can do. The errors here are more nicely split and benefit from the module separation, even if it's a new way.

Reply:
Ignore my internal monologue, I'll send this through later today!